lib/screens/v2/widget_instance/alert.ex

```diff
@@ -17,30 +17,59 @@ defmodule Screens.V2.WidgetInstance.Alert do
           screen: Screens.Config.Screen.t(),
           alert: Alert.t(),
           stop_sequences: list(list(stop_id())),
-          routes_at_stop: list(%{route_id: route_id(), active?: boolean()})
+          routes_at_stop: list(%{route_id: route_id(), active?: boolean()}),
+          now: DateTime.t()
         }
 
-  def active?(%__MODULE__{alert: alert}, happening_now? \\ &Alert.happening_now?/1) do
-    happening_now?.(alert)
+  @base_priority 2
+
+  @effect_priorities ~w[shuttle stop_closure suspension station_closure detour stop_moved snow_route elevator_closure]a
+                     |> Enum.with_index(1)
+
+  @spec priority(t()) :: nonempty_list(pos_integer()) | :no_render
+  def priority(t) do
+    tiebreakers = [
+      @base_priority,
+      tiebreaker_primary_timeframe(t),
+      tiebreaker_location(t),
+      tiebreaker_secondary_timeframe(t),
+      tiebreaker_effect(t)
+    ]
+
+    if Enum.any?(tiebreakers, &(&1 == :no_render)), do: :no_render, else: tiebreakers
   end
 
-  @spec seconds_from_onset(t(), DateTime.t()) :: integer() | :infinity
-  def seconds_from_onset(t, now \\ DateTime.utc_now())
+  def serialize(_t) do
+    %{}
+  end
 
-  def seconds_from_onset(%__MODULE__{alert: %Alert{active_period: [{start, _} | _]}}, now)
-      when not is_nil(start) do
-    DateTime.diff(now, start, :second)
+  def slot_names(_t) do
+    [:medium_left, :medium_right]
+  end
+
+  def widget_type(_t) do
+    :alert
+  end
+
+  def valid_candidate?(t) do
+    priority(t) != :no_render and slot_names(t) != :no_render
   end
 
-  def seconds_from_onset(_t, _now), do: :infinity
+  def active?(%__MODULE__{alert: alert, now: now}, happening_now? \\ &Alert.happening_now?/2) do
+    happening_now?.(alert, now)
+  end
 
-  @spec seconds_to_next_active_period(t(), DateTime.t()) :: integer() | :infinity
-  def seconds_to_next_active_period(t, now \\ DateTime.utc_now())
+  @spec seconds_from_onset(t()) :: integer()
+  def seconds_from_onset(%__MODULE__{alert: %Alert{active_period: [{start, _} | _]}, now: now})
+      when not is_nil(start) do
```

Comment:
Is there a fallback in case `start` is `nil` or does Elixir have something like that by default?

Reply:
There is not (the function would raise an exception in that case), but when I checked all alerts in the system, their active periods always had a non-null start value.

The API documentation also seems to indicate that the start value is non-nullable. I can't deep-link it, but if you scroll to the "Models" section at the bottom of the [V3 API docs](https://api-v3.mbta.com/docs/swagger/index.html), the ActivePeriod type shows only the end date as nullable:
<img width="436" alt="image" src="https://user-images.githubusercontent.com/63608771/123313536-9c58ad80-d4f7-11eb-937e-49a3d14f2d69.png">